medvis/dicom/src/errorlogging.cpp

```diff
@@ -0,0 +1,240 @@
+    static ivw_gdcm_log_stream<LogLevel::Error> error_log_stream;
+
+    // reset logging/tracing in gdcm
+    gdcm::Trace::Trace();
+
+    switch (verbosity) {
+        case LogVerbosity::Info:
+            gdcm::Trace::DebugOn();
+            gdcm::Trace::SetDebugStream(debug_log_stream);
+            [[fallthrough]];
+        case LogVerbosity::Warn:
+            gdcm::Trace::WarningOn();
+            gdcm::Trace::SetWarningStream(warning_log_stream);
+            [[fallthrough]];
+        case LogVerbosity::Error:
+            gdcm::Trace::ErrorOn();
+            gdcm::Trace::SetErrorStream(error_log_stream);
+            break;
+        case LogVerbosity::None:
+        default:
+            break;
+    }
+
+    /*
+#if defined(IVW_DEBUG)
+    gdcmDebugMacro("GDCM debug output available.");
+    gdcmWarningMacro("GDCM warning output available.");
+    gdcmErrorMacro("GDCM error output available.");
+#endif
+    */
+}
+
+// =====================================================================
+// logging functionality of libtiff - using variable argument lists and snprintf
+// =====================================================================
+
+// sprintf for msvc - http://stackoverflow.com/questions/2915672/snprintf-and-visual-studio-2010
+#if defined(_MSC_VER) && _MSC_VER < 1900
+#define snprintf c99_snprintf
+#define vsnprintf c99_vsnprintf
+
+inline int c99_vsnprintf(char *outBuf, size_t size, const char *format, va_list args) {
+    int count = -1;
+
+    if (size != 0) count = _vsnprintf_s(outBuf, size, _TRUNCATE, format, args);
+    if (count == -1) count = _vscprintf(format, args);
+
+    return count;
+}
+
+inline int c99_snprintf(char *outBuf, size_t size, const char *format, ...) {
+    int count;
+    va_list args;
+
+    va_start(args, format);
+    count = c99_vsnprintf(outBuf, size, format, args);
+    va_end(args);
+
+    return count;
+}
+#endif
+
+void tiffErrorHandler(const char *module, const char *fmt, va_list args) {
+    char buffer[512];
+    int res = vsnprintf(buffer, 512, fmt, args);
+
+    std::string err_str;
+    if (res < 1) {
+        err_str = std::string("got libtiff error from ") + module + " but snprintf failed.";
+    } else {
+        err_str = buffer;
+    }
+
+    inviwo::LogCentral::getPtr()->log("Gdcm Volume Importer", LogLevel::Error, LogAudience::User,
+                                      "<libtiff>", module, 0, err_str);
+}
+
+void tiffWarningHandler(const char *module, const char *fmt, va_list args) {
+    char buffer[512];
+    int res = snprintf(buffer, 512, fmt, args);
+
+    std::string err_str;
+    if (res < 1) {
+        err_str = std::string("got libtiff warning from ") + module + " but snprintf failed.";
+    } else {
+        err_str = buffer;
+    }
+
+    inviwo::LogCentral::getPtr()->log("Gdcm Volume Importer", LogLevel::Warn, LogAudience::User,
+                                      "<libtiff>", module, 0, err_str);
+}
+
+void enableTiffLogging(LogVerbosity verbosity) {
```

Comment:
Do we do this in our tiff stuff?

Reply:
Not as of now. Question is, which module should provide this functionality?